app/src/main/java/io/apicurio/registry/ccompat/store/RegistryStorageFacade.java

```diff
@@ -0,0 +1,41 @@
+package io.apicurio.registry.ccompat.store;
+
+import io.apicurio.registry.ccompat.dto.RegisterSchemaRequest;
+import io.apicurio.registry.ccompat.dto.Schema;
+import io.apicurio.registry.storage.ArtifactAlreadyExistsException;
+import io.apicurio.registry.storage.ArtifactNotFoundException;
+import io.apicurio.registry.storage.RegistryStorageException;
+import io.apicurio.registry.storage.VersionNotFoundException;
+
+import java.util.List;
+import java.util.Set;
+import java.util.SortedSet;
+
+/**
+ * @author Ales Justin
+ */
+public interface RegistryStorageFacade {
+    Set<String> listSubjects();
+
+    SortedSet<Long> deleteSubject(String subject) throws ArtifactNotFoundException, RegistryStorageException;
+
+    String getSchema(Integer id) throws ArtifactNotFoundException, RegistryStorageException;;
+
+    Schema getSchema(String subject, String version) throws ArtifactNotFoundException, VersionNotFoundException, RegistryStorageException;;
+
+    List<Integer> listVersions(String subject) throws ArtifactNotFoundException, RegistryStorageException;
+
+    Schema findSchemaWithSubject(String subject, boolean checkDeletedSchema, String schema) throws ArtifactNotFoundException, RegistryStorageException;;
+
+    /**
+     * @return global id
+     */
+    int registerSchema(String subject, Integer id, Integer version, String schema) throws ArtifactAlreadyExistsException, ArtifactNotFoundException, RegistryStorageException;
+
+    /**
+     * @return schema version as integer
+     */
+    int deleteSchema(String subject, String version) throws ArtifactNotFoundException, VersionNotFoundException, RegistryStorageException;
+
+    boolean testCompatibility(String subject, String version, RegisterSchemaRequest request) throws ArtifactNotFoundException, VersionNotFoundException, RegistryStorageException;
```

Comment:
Should test-compatibility be on the storage facade?  Not sure the storage will be testing that.

Reply:
Ah, true.
This was a leftover from old code.
I'll remove this today, as I'll put in place validation/compatibility/rules interface.